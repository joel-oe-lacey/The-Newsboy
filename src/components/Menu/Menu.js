import React from 'react';
import './Menu.scss'

const Menu = (props) => {
    const selectNews = (event, type) => {
        event.preventDefault();
        props.selectNewsType(type);
    }

    return (
        <nav className="menu">
            <h1>What's <span className="headerColor">New?</span></h1>
            <button onClick={(event) => selectNews(event, 'local')}>Local News</button>
            <button onClick={(event) => selectNews(event, 'technology')}>Technology</button>
            <button onClick={(event) => selectNews(event, 'entertainment')}>Entertainment</button>
            <button onClick={(event) => selectNews(event, 'science')}>Science</button>
            <button onClick={(event) => selectNews(event, 'health')}>Health</button>
        </nav>
    )
}

export default Menu;