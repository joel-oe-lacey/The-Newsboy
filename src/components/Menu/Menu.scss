.menu {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-top: 2%;
    grid-area: menu;

    button {
        height: 5%;
        width: 100%;
    }
}