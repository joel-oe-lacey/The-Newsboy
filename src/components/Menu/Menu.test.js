import React from 'react';
import { shallow } from 'enzyme';
import Menu from './Menu';

describe('Menu', () => {
  it('renders the baseline menu structure', () => {
    const wrapper = shallow(<Menu selectNewsType={jest.fn()} />);

    expect(wrapper).toMatchSnapshot();
  })
})