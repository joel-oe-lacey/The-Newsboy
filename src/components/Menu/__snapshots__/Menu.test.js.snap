// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders the baseline menu structure 1`] = `ShallowWrapper {}`;
